src/web/components/Lazy.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { css } from 'emotion';
+
+import { useHasBeenSeen } from '@root/src/web/lib/useHasBeenSeen';
+
+type Props = {
+    children: JSX.Element;
+    margin: number;
+};
+
+// Ensure the ref wrapper expands. This is used for componenents like
+// MostViewedRightWrapper that needs to check it's parent's height
+const flexGrowStyles = css`
+    display: flex;
+    flex-grow: 1;
+`;
+
+export const Lazy = ({ children, margin }: Props) => {
+    const [hasBeenSeen, setRef] = useHasBeenSeen({
+        rootMargin: `${margin}px`,
+    });
+
+    if (typeof setRef !== 'function') {
+        return null;
+    }
```

Comment:
This is needed to keep typescript happy

Reply:
could you not add it to `useHasBeenSeen<[boolean, function]>` ?

Reply:
I'd like to but couldn't make it work. If you want to have a go, be my hero!

Reply:
^^ I tried and failed :P it would be good to add a comment as to why its there withe a little explainer